bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewHubHandler.java

```diff
@@ -89,7 +87,7 @@ public class HDPowerViewHubHandler extends BaseBridgeHandler {
     private long hardRefreshPositionInterval;
     private long hardRefreshBatteryLevelInterval;
 
-    private @Nullable HDPowerViewWebTargets webTargets;
+    private @NonNullByDefault({}) HDPowerViewWebTargets webTargets;
```

Comment:
Yes, it is. It cancels the default for this field, so effectively it becomes non-annotated. This is a little cheating, but can be used for cases like this where we know initialization is ensured by the framework. Strict `@NonNull` annotation will not work, since we would then have to initialize before leaving the constructor. And `@Nullable` enforces null checks everywhere, although we know this is not needed for any methods called after `initialize()`.

If you search the repository, you will find this trick used in many similar cases, so it's not unusual or discouraged when used with care. It's also common to use in test classes when having initialization in `@BeforeEach`. Situation is similar, the test framework will call this method, so after that there is no reason for null checks.

Reply:
Ah. Very interesting. Many thanks for the info.